homeassistant/components/metoffice/sensor.py

```diff
@@ -30,175 +32,147 @@
+        "mdi:thermometer",
+    ],
+    "wind_speed": ["Wind Speed", SPEED_MILES_PER_HOUR, "mdi:weather-windy"],
+    "wind_direction": ["Wind Direction", None, "mdi:compass-outline"],
+    "wind_gust": ["Wind Gust", SPEED_MILES_PER_HOUR, "mdi:weather-windy"],
+    "visibility": ["Visibility", None, "mdi:eye"],
+    "visibility_distance": ["Visibility Distance", LENGTH_KILOMETERS, "mdi:eye"],
+    "uv": ["UV Index", UV_INDEX, "mdi:weather-sunny-alert"],
+    "precipitation": [
+        "Probability of Precipitation",
+        UNIT_PERCENTAGE,
+        "mdi:weather-rainy",
+    ],
+    "humidity": ["Humidity", UNIT_PERCENTAGE, "mdi:water-percent"],
 }
 
-PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
-    {
-        vol.Required(CONF_API_KEY): cv.string,
-        vol.Required(CONF_MONITORED_CONDITIONS, default=[]): vol.All(
-            cv.ensure_list, [vol.In(SENSOR_TYPES)]
-        ),
-        vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
-        vol.Inclusive(
-            CONF_LATITUDE, "coordinates", "Latitude and longitude must exist together"
-        ): cv.latitude,
-        vol.Inclusive(
-            CONF_LONGITUDE, "coordinates", "Latitude and longitude must exist together"
-        ): cv.longitude,
-    }
-)
-
-
-def setup_platform(hass, config, add_entities, discovery_info=None):
-    """Set up the Met Office sensor platform."""
-    api_key = config.get(CONF_API_KEY)
-    latitude = config.get(CONF_LATITUDE, hass.config.latitude)
-    longitude = config.get(CONF_LONGITUDE, hass.config.longitude)
-    name = config.get(CONF_NAME)
-
-    datapoint = dp.connection(api_key=api_key)
-
-    if None in (latitude, longitude):
-        _LOGGER.error("Latitude or longitude not set in Home Assistant config")
-        return
-
-    try:
-        site = datapoint.get_nearest_site(latitude=latitude, longitude=longitude)
-    except dp.exceptions.APIException as err:
-        _LOGGER.error("Received error from Met Office Datapoint: %s", err)
-        return
-
-    if not site:
-        _LOGGER.error("Unable to get nearest Met Office forecast site")
-        return
 
-    data = MetOfficeCurrentData(hass, datapoint, site)
-    data.update()
-    if data.data is None:
-        return
+async def async_setup_entry(
+    hass: HomeAssistantType, entry: ConfigType, async_add_entities
+) -> None:
+    """Set up the Met Office weather sensor platform."""
+    hass_data = hass.data[DOMAIN][entry.entry_id]
 
-    sensors = []
-    for variable in config[CONF_MONITORED_CONDITIONS]:
-        sensors.append(MetOfficeCurrentSensor(site, data, variable, name))
-
-    add_entities(sensors, True)
+    async_add_entities(
+        [
+            MetOfficeCurrentSensor(entry.data, hass_data, sensor_type)
+            for sensor_type in SENSOR_TYPES
+        ],
+        False,
+    )
 
 
 class MetOfficeCurrentSensor(Entity):
-    """Implementation of a Met Office current sensor."""
+    """Implementation of a Met Office current weather condition sensor."""
```

Comment:
Does it make sense to use the available property with the coordinator status?

Reply:
Agreed. Done.